src/deepsparse/utils/timer.py

```diff
@@ -338,6 +338,16 @@ def all_times(self) -> Dict[str, List[float]]:
 
         return all_times
 
+    def current_or_new(self) -> StagedTimer:
+        """
+        Return the current timer if there is one, otherwise return a new one.
+        """
+        if self.current:
+            return self.current
+        else:
+            with self.new_timer_context(total_inference=False) as timer:
+                return timer
```

Comment:
is this tested? just want to make sure it won't exit the context and leave with current not set (even if the timer object still exists)

Reply:
Right, I'm sure it will work, only question I had is if the current that is created will actually live beyond the context in this function

Reply:
Good call, this is worth investigating in a followup